synapse/models/geospace.py

```diff
@@ -1,85 +1,109 @@
+import struct
+
+import synapse.exc as s_exc
 import synapse.lib.gis as s_gis
 import synapse.lib.types as s_types
+import synapse.lib.module as s_module
 import synapse.lib.syntax as s_syntax
 
-import synapse.lib.module as s_module
+units = {
+    'mm': 1,
+    'cm': 10,
+
+    'm': 1000,
+    'meters': 1000,
 
-class LatLongType(s_types.DataType):
+    'km': 1000000,
+}
+
+
+class LatLong(s_types.Type):
+    # FIXME nothing uses these props
+    # FIXME should these be indexed as something other than a string?
+
+    def postTypeInit(self):
+        self.setNormFunc(str, self._normPyStr)
 
-    def norm(self, valu, oldval=None):
+    def _normPyStr(self, valu):
 
         valu = valu.strip().lower()
         if valu == '??':
             return valu, {}
 
         try:
+            latv, lonv = valu.split(',', 1)
+            latv, lonv = float(latv.strip()), float(lonv.strip())
+        except Exception as e:
+            raise s_exc.BadTypeValu(valu, mesg='Invalid float format')
 
-            lat, lon = valu.split(',', 1)
-
-            latv = float(lat.strip())
-            lonv = float(lon.strip())
+        if latv > 90 or latv < -90:
+            raise s_exc.BadTypeValu(valu, mesg='Latitude may only be -90.0 to 90.0')
+        if lonv > 180 or lonv < -180:
+            raise s_exc.BadTypeValu(valu, mesg='Longitude may only be -180.0 to 180.0')
 
-        except Exception as e:
-            self._raiseBadValu(valu, mesg='Invalid float format')
+        return (latv, lonv), {'subs': {'lat': latv, 'lon': lonv}}
 
-        #TODO eventually support minutes / sec and N/S E/W syntax
+    def indx(self, valu):
+        # to unpack:
+        # latv, lonv = struct.unpack('>qq', norm)
+        # latv, lonv = latv / 10**8, lonv / 10**8
+        return struct.pack('>qq', int(valu[0] * 10**8), int(valu[1] * 10**8))
```

Comment:
There's still a trick here about signdness.  btree range indexing over signed values requires a scale up.  lets chat in a go-back tomorrow :)